optionlab/engine.py

```diff
@@ -177,321 +168,96 @@ def run(self):
             An Outputs object containing the output of a calculation.
         """
 
-        time2target = self.__days2target / self.__daysinyear
-        self.cost = [0.0 for _ in range(len(self.__type))]
+        time_to_target = self.days_to_target / self._days_in_year
+        self.cost = [0.0] * len(self.type)
         self.implied_volatility = []
         self.itm_probability = []
         self.delta = []
         self.gamma = []
         self.vega = []
         self.theta = []
 
-        if self.__s.shape[0] == 0:
-            self.__s = create_price_seq(self.__minstock, self.__maxstock)
-
-        self.profit = zeros((len(self.__type), self.__s.shape[0]))
-        self.strategyprofit = zeros(self.__s.shape[0])
-
-        if self.__compute_expectation and self.__s_mc.shape[0] == 0:
-            self.__s_mc = create_price_samples(
-                self.__stockprice,
-                self.__volatility,
-                time2target,
-                self.__r,
-                self.__distribution,
-                self.__y,
-                self.__nmcprices,
+        self.profit = zeros((len(self.type), self.s.shape[0]))
+        self.strategy_profit = zeros(self.s.shape[0])
+
+        if self._compute_expectation and self.terminal_stock_prices.shape[0] == 0:
+            self.terminal_stock_prices = create_price_samples(
```

Comment:
When a module with other option pricing models is created, this create_price_samples() will have to be renamed and moved there.